apps/speed.c

```diff
@@ -604,6 +610,16 @@ static OPT_PAIR eddsa_choices[] = {
 # define EdDSA_NUM       OSSL_NELEM(eddsa_choices)
 
 static double eddsa_results[EdDSA_NUM][2];    /* 2 ops: sign then verify */
+
+# ifndef OPENSSL_NO_SM2
+#  define R_EC_SM2P256        0
+static OPT_PAIR sm2_choices[] = {
+    {"sm2p256", R_EC_SM2P256}
```

Comment:
Where does the p256 bit come from in sm2p256?

Reply:
SM2 uses elliptic curves over 256-bit prime fields, so I think it's proper to say something like SM2P256...

Reply:
Is it commonly referred to by that name? I'd rather not invent a new curve name. My understanding was that although SM2 can be used with other curves, it never normally is. So "SM2" to me means, "SM2 using the default SM2 curve" - and you'd only specify the curve if it was something else. P256 on the other hand reminds me of the NIST curve of that name.

Reply:
> Is it commonly referred to by that name?

No, this is the first time that name appeared in all history I think...

> My understanding was that although SM2 can be used with other curves, it never normally is. So "SM2" to me means, "SM2 using the default SM2 curve"

Right, so how about using "curveSM2" which is defined in https://tools.ietf.org/html/draft-yang-tls-tls13-sm-suites-01#section-4, and we are standardizing that name along with the whole draft ;-)

Reply:
"curveSM2" works for me!

Reply:
Great, all comments have been addressed!